Libraries/xcsdk/Tools/xcrun.cpp

```diff
@@ -236,15 +241,25 @@ Version()
 int
 main(int argc, char **argv)
 {
-    std::vector<std::string> args = std::vector<std::string>(argv + 1, argv + argc);
-
     /*
      * Parse out the options, or print help & exit.
      */
     Options options;
-    std::pair<bool, std::string> result = libutil::Options::Parse<Options>(&options, args);
-    if (!result.first) {
-        return Help(result.second);
+
+    {
+        std::vector<std::string> args = std::vector<std::string>(argv + 1, argv + argc);
+        std::string xcrunBaseName = libutil::FSUtil::GetBaseName(argv[0]);
+        /* HACK(strager): There must be a better way. */
+        bool isXcrun = xcrunBaseName == "xcrun";
+        if (isXcrun) {
+            std::pair<bool, std::string> result = libutil::Options::Parse<Options>(&options, args);
+            if (!result.first) {
+                return Help(result.second);
+            }
+        } else {
+            options.tool() = std::move(xcrunBaseName);
```

Comment:
Modifying the options struct is a little unfortunate. Maybe instead make local variable for each that is assigned differently based on the condition above, then change the code below to use that variable instead of `options`?